File: src/main/java/model/criteria/BaseCriteria.java
package model.criteria;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class BaseCriteria {
    private GenderCriteria genderCriteria;
    private LifestyleCriteria lifestyleCriteria;
    private Integer ageCriteria;
    private Integer ageOffset;
    private RoomTypeCriteria roomTypeCriteria;
}

File: flatm8-ui/src/app/model/BaseCriteria.ts
export class BaseCriteria {
  genderCriteria: string;
  lifestyleCriteria: string;
  ageCriteria: number;
  ageOffset: number;
  roomTypeCriteria: string;

  constructor (private ageOffsetDefault: number) {
    this.ageOffset = ageOffsetDefault;
  }
}
